### ISSUE TITLE
kafkaconnector.https.CamelHttpsSinkConnector oauth2 scope is missing

### ISSUE THREAD

**Issue by chethan-bs** [apache/camel-kafka-connector#1647]
i am using  CamelHttpsSinkConnector REST API with Oauth2 Client and i see scope is not exists on the code in order to use oauth2TokenEndpoint /oauth2/v2.0/token from azure. below is the config where i need oauth2Scope to be defined.
 
        "connector.class": "org.apache.camel.kafkaconnector.https.CamelHttpsSinkConnector",
        "camel.sink.endpoint.okStatusCodeRange": "200,201",
        "topics": "msk.databook.test.openlineage.topic.internal.any.v1",

        "camel.sink.endpoint.httpMethod": "POST",
        "camel.sink.path.httpUri": "<url>",
        "camel.sink.endpoint.oauth2ClientId": "<client_id>",
        "camel.sink.endpoint.oauth2ClientSecret": "<client_secret>",
        "camel.sink.endpoint.oauth2TokenEndpoint": "https://login.microsoftonline.com/<sub>/oauth2/v2.0/token",
        **"camel.sink.endpoint.oauth2Scope": "api://<client_id>/.default",**

**Comment by oscerd** [comment]
I created this on Kamelets catalog side: https://github.com/apache/camel-kamelets/issues/2062

**Comment by oscerd** [comment]
No, you need to add the parameter here: https://github.com/apache/camel/blob/main/components/camel-http-common/src/main/java/org/apache/camel/http/common/HttpConfiguration.java#L41, but first we need to create an issue in Camel JIRA https://issues.apache.org/jira/browse/CAMEL

**Comment by davsclaus** [comment]
https://issues.apache.org/jira/browse/CAMEL-20924

**Comment by davsclaus** [comment]
This comes in Camel 4.7 onwards

**Comment by oscerd** [comment]
This will be in Camel-Kamelets 4.8.0

**Comment by oscerd** [comment]
This should be ok now
